assembly/src/ast/mod.rs

```diff
@@ -201,7 +223,19 @@ impl ProgramAst {
 
         let local_procs = sort_procs_into_vec(context.local_procs);
         let (nodes, locations) = body.into_parts();
-        Ok(Self::new(nodes, local_procs, imports)?.with_source_locations(locations, start))
+
+        if used_imported_procs.len() > MAX_USED_IMPORTED_PROCS {
+            return Err(ParsingError::too_many_imported_procs_used(
+                used_imported_procs.len(),
+                MAX_USED_IMPORTED_PROCS,
+            ));
+        }
```

Comment:
I wonder if it would be better to move this check inside `ModuleImports` somehow. If we change `parse_imports()` to return `ModuleImports`, we could also check if the valid number of used procedure calls was made inside `add_used_imported_proc()` (and return `ParsingError` if not).

Reply:
Apart from the mutability issues which may prevent this altogether, we would end up performing the check every time a new procedure use was added to the map. The check is currently only performed once, after parsing is complete.

Reply:
I think this is OK. The check is not super expensive and in general parsing performance is not super important.